components/builder-depot/src/server.rs

```diff
@@ -841,9 +849,11 @@ fn upload_package(req: &mut Request) -> IronResult<Response> {
             }
         };
 
-    if origin_package_found && depot.archive(&ident, &target_from_artifact).is_some() {
+    // Using the S3 backend if package exists in originsrv
+    // we make the assumption that the package exists in s3
+    if origin_package_found {
```

Comment:
You can return directly from inside the match, and get rid of the origin_package_found variable entirely..

Reply:
Oh duh, excellent catch!